### PR TITLE
Introduce fishForSpecificMessage which returns properly typed message

### LINKED ISSUE #20118: Alternative fishForMessage that returns result of partial function
Akka TestKit contains a method with this signature:

``` scala
def fishForMessage(max: Duration = Duration.Undefined, hint: String = "")(f: PartialFunction[Any, Boolean]): Any
```

Is there a reason there's no alternative version which looks like this:

``` scala
def fishForMessage[T](max: Duration = Duration.Undefined, hint: String = "")(f: PartialFunction[Any, T]): T
```

### PR DESCRIPTION
**Pull request by miciek**
Resolves #20118

### CODE REVIEW

File: akka-testkit/src/main/scala/akka/testkit/TestKit.scala [hunk 1]
```diff
@@ -440,6 +440,21 @@ trait TestKitBase {
   }
 
   /**
+   * Same as `fishForMessage`, but gets a different partial function and returns properly typed message.
+   */
+  def fishForSpecificMessage[T](max: Duration = Duration.Undefined, hint: String = "")(f: PartialFunction[Any, T]): T = {
```
Comment (patriknw) [inline akka-testkit/src/main/scala/akka/testkit/TestKit.scala]: good name

**Review by patriknw — APPROVED** [review]
LGTM, thanks

**Comment by patriknw** [comment]
OK TO TEST

**Comment by miciek** [comment]
Can someone review it?

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

**Comment by johanandren** [comment]
PLS BUILD

**Comment by miciek** [comment]
Anybody knows why this is not binary compatible?

**Comment by johanandren** [comment]
@miciek 
```
[info] akka-testkit: found 3 potential binary incompatibilities while checking against com.typesafe.akka:akka-testkit_2.11:2.4.11 
[error]  * synthetic method fishForSpecificMessage$default$1()scala.concurrent.duration.Duration in trait akka.testkit.TestKitBase is present only in current version
[error]    filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.testkit.TestKitBase.fishForSpecificMessage$default$1")
[error]  * method fishForSpecificMessage(scala.concurrent.duration.Duration,java.lang.String,scala.PartialFunction)java.lang.Object in trait akka.testkit.TestKitBase is present only in current version
[error]    filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.testkit.TestKitBase.fishForSpecificMessage")
[error]  * synthetic method fishForSpecificMessage$default$2()java.lang.String in trait akka.testkit.TestKitBase is present only in current version
[error]    filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.testkit.TestKitBase.fishForSpecificMessage$default$2")
```

We don't maintain binary comp for the testkit though, so you can safely add those filters to `project/MiMa.scala`.

**Comment by miciek** [comment]
@johanandren Thanks. Added the filters needed. Please build again.

**Comment by johanandren** [comment]
Our CI will build automagically when you push a new commit. There may be a long line of jobs waiting though, so it may take a while.

**Pull request merged**